### PR TITLE
Factory method

### PR DESCRIPTION
**Pull request by Lenkelheit**

### CODE REVIEW

File: Task1/Project/Shapes/Models/Classes/Circle.cs [hunk 1]
```diff
@@ -38,7 +39,7 @@ public Circle(Point сenter, double radius)
         /// <summary>
         /// Identifier of the circle.
         /// </summary>
-        public override string ID => nameof(Circle);
+        public override string ID => ShapeBase.IdOfTheShape(this.GetType());
```
Comment (iamprovidence) [inline Task1/Project/Shapes/Models/Classes/Circle.cs]: you do not need ID in all child classes, just write it in base class once, that would be enough

File: Task1/Project/Shapes/Models/Classes/ShapeBase.cs [hunk 2]
```diff
@@ -8,8 +11,25 @@ namespace Shapes.Models.Classes
     /// </summary>
     public abstract class ShapeBase : IShape, IFileManager
     {       
+        // FIELDS
+        static IdentifierShape idOfTheShape;
```
Comment (iamprovidence) [inline Task1/Project/Shapes/Models/Classes/ShapeBase.cs]: it is not **idOfTheShape**, it is a method that create **ID** by Type

File: Task1/Project/Shapes/Models/Classes/ShapeBase.cs [hunk 3]
```diff
@@ -8,8 +11,25 @@ namespace Shapes.Models.Classes
     /// </summary>
     public abstract class ShapeBase : IShape, IFileManager
     {       
+        // FIELDS
+        static IdentifierShape idOfTheShape;
+        static Dictionary<string, Deserialization> dictionaryOfId;
```
Comment (iamprovidence) [inline Task1/Project/Shapes/Models/Classes/ShapeBase.cs]: it is not **dictionaryOfId**, it is a **factories**, something that by key can create object

File: Task1/Project/Shapes/Models/Classes/ShapeBase.cs [hunk 4]
```diff
@@ -8,8 +11,25 @@ namespace Shapes.Models.Classes
...
     {       
+        // FIELDS
+        static IdentifierShape idOfTheShape;
+        static Dictionary<string, Deserialization> dictionaryOfId;    
         // PROPERTIES
-         /// <summary>
+        /// <summary>
+        /// The identifier of the shape.
+        /// </summary>
+        public static IdentifierShape IdOfTheShape
```
Comment (iamprovidence) [inline Task1/Project/Shapes/Models/Classes/ShapeBase.cs]: Bad naming and Documentation, was mentioned above

File: Task1/Project/Shapes/Models/Classes/ShapeBase.cs [hunk 5]
```diff
@@ -41,6 +61,12 @@ public CoordinateQuarters GetQuarter
                 else return middlePoint.Y > 0 ? CoordinateQuarters.Second : CoordinateQuarters.Third;
             }
         }
+        // CONSTRUCTOR
+        static ShapeBase()
+        {
+            dictionaryOfId = new Dictionary<string, Deserialization>();
+            idOfTheShape = (type) => type.Name;
```
Comment (iamprovidence) [inline Task1/Project/Shapes/Models/Classes/ShapeBase.cs]: you should also registrate all your known classes here

File: Task1/Project/Shapes/Models/Classes/ShapeBase.cs [hunk 6]
```diff
@@ -50,89 +76,89 @@ public CoordinateQuarters GetQuarter
...
-        /// <param name="line">
-        /// The string data.
+        /// <param name="key">
+        /// Id of the shape.
         /// </param>
-        protected abstract void Interpret(string line);       
+        /// <param name="transformer">
+        /// Transformer <see cref="string"/> to <see cref="ShapeBase"/>.
+        /// </param>
+        public static void RegisterShape(string key, Deserialization transformer)
```
Comment (iamprovidence) [inline Task1/Project/Shapes/Models/Classes/ShapeBase.cs]: should take a **System.Type** to registrate as a first argument

Comment (iamprovidence) [reply Task1/Project/Shapes/Models/Classes/ShapeBase.cs]: and it is not a transformer, its probably more as creator, or deserializator

File: Task1/Project/Shapes/Models/Classes/Circle.cs (hunk 1 above)
Comment (Lenkelheit) [reply Task1/Project/Shapes/Models/Classes/Circle.cs]: Ok.

File: Task1/Project/Shapes/Models/Classes/ShapeBase.cs (hunk 2 above)
Comment (Lenkelheit) [reply Task1/Project/Shapes/Models/Classes/ShapeBase.cs]: It has just done.

File: Task1/Project/Shapes/Models/Classes/ShapeBase.cs (hunk 4 above)
Comment (Lenkelheit) [reply Task1/Project/Shapes/Models/Classes/ShapeBase.cs]: Done.

File: Task1/Project/Shapes/Models/Classes/ShapeBase.cs (hunk 3 above)
Comment (Lenkelheit) [reply Task1/Project/Shapes/Models/Classes/ShapeBase.cs]: Rewrote.

File: Task1/Project/Shapes/Models/Classes/ShapeBase.cs (hunk 6 above)
Comment (Lenkelheit) [reply Task1/Project/Shapes/Models/Classes/ShapeBase.cs]: I wrote "creator". It will be normal, I think.

Comment (iamprovidence) [reply Task1/Project/Shapes/Models/Classes/ShapeBase.cs]: it shall be

File: Task1/Project/Shapes/Models/Classes/ShapeBase.cs (hunk 5 above)
Comment (Lenkelheit) [reply Task1/Project/Shapes/Models/Classes/ShapeBase.cs]: Done.

**Pull request merged**